jira-list-functions.js

```diff
@@ -0,0 +1,166 @@
+const { getJiraClient } = require("./helpers");
+
+const DEFAULT_FIELDS = ["created", "description", "summary", "status", "priority"];
+const DEFAULT_MAX_RESULTS = 50;
+const DEFAULT_START_AT = 0;
+
+function listIssues({
+  host,
+  email,
+  apiToken,
+  project,
+  status,
+  startDate,
+  endDate,
+  overrideJql: customJqlString,
+  maxResults = DEFAULT_MAX_RESULTS,
+  startAt = DEFAULT_START_AT,
+  fields = DEFAULT_FIELDS,
+}) {
+  const jiraClient = getJiraClient({ host, email, apiToken });
+
+  const jqlSegments = [];
+  if (customJqlString) {
+    jqlSegments.push(customJqlString);
+  }
+  if (!customJqlString && project) {
+    jqlSegments.push(`project = ${project}`);
+  }
+  if (!customJqlString && status) {
+    jqlSegments.push(`status = ${status}`);
+  }
+  if (!customJqlString && startDate) {
+    const startDateFormatted = new Date(startDate).toISOString().split("T")[0];
```

Comment:
Every such call in this file could go to `helpers.js`, which would help if we decided to alter the date parsing in future`;